src/explore-education-statistics-common/src/components/CopyTextButton.tsx

```diff
@@ -3,89 +3,70 @@ import styles from '@common/components/CopyTextButton.module.scss';
-  label?: string;
+  id: string;
+  inline?: boolean;
+  label: string;
   labelHidden?: boolean;
   text: string;
 }
 
 export default function CopyTextButton({
   buttonText = 'Copy',
   className,
-  confirmMessage = 'Text copied to the clipboard.',
-  inlineButton = true,
+  confirmMessage = 'Copied',
+  id,
+  inline = true,
   label,
-  labelHidden = true,
+  labelHidden,
   text,
 }: CopyTextButtonProps) {
   const [copied, toggleCopied] = useToggle(false);
 
+  useEffect(() => {
+    const resetTimeout = setTimeout(toggleCopied.off, 5000);
+
+    return () => {
+      if (copied) {
+        clearTimeout(resetTimeout);
+      }
+    };
+  }, [copied, toggleCopied]);
+
   return (
-    <div className={className}>
-      <div
-        className={classNames(styles.container, {
-          'dfe-flex dfe-align-items-start': inlineButton,
+    <div
+      className={classNames(className, styles.container, {
+        'dfe-flex dfe-align-items--start': inline,
+      })}
+    >
+      <UrlContainer
+        className={classNames(styles.urlContainer, {
+          'dfe-flex-grow--1': inline,
+          'govuk-!-margin-bottom-2': !inline,
         })}
-      >
-        <UrlContainer
-          id="copy-link-url"
-          className={styles.urlContainer}
-          label={label}
-          labelHidden={labelHidden}
-          url={text}
-          widthLimited={false}
-        />
-        <div
-          className={classNames({
-            'dfe-flex dfe-align-items--center govuk-!-margin-top-2':
-              !inlineButton,
-          })}
-        >
-          <Button
-            className={classNames('govuk-!-margin-bottom-0', {
-              'govuk-!-margin-right-2': !inlineButton,
-            })}
-            onClick={async () => {
-              await navigator.clipboard.writeText(text);
-              toggleCopied.on();
-            }}
-          >
-            {buttonText}
-          </Button>
-          {!inlineButton && (
-            <Message copied={copied} confirmMessage={confirmMessage} />
-          )}
-        </div>
-      </div>
-      {inlineButton && (
-        <Message
-          className={styles.message}
-          copied={copied}
-          confirmMessage={confirmMessage}
-        />
-      )}
-    </div>
-  );
-}
+        id={id}
+        inline={inline}
+        label={label}
+        labelHidden={labelHidden}
+        url={text}
+      />
 
-function Message({
-  className,
-  copied,
-  confirmMessage,
-}: {
-  className?: string;
-  copied: boolean;
-  confirmMessage?: string;
-}) {
-  return (
-    <div aria-live="polite" className={className}>
-      {copied && confirmMessage}
+      <Button
+        className={classNames(styles.button, {
+          'govuk-!-margin-right-2': !inline,
+        })}
+        onClick={async () => {
+          await navigator.clipboard.writeText(text);
+          toggleCopied.on();
+        }}
+      >
+        <span aria-live="polite">{copied ? confirmMessage : buttonText}</span>
```

Comment:
I think only the confirmMessage should be wrapped with an aria-live, as buttonText need not be notifed to users, I think the current setup would announce the confirm message, then announce the buttonText 5 seconds later

Reply:
Good point! I've updated this to use `ScreenReaderMessage` instead (which should handle the behaviour correct:

![image](https://github.com/user-attachments/assets/85c746e4-24f4-4af9-acd5-e0ac293156bd)